File: code-awesome/java-quarkus-angular/super_dating/backend/src/pages/task.java
package pages;

import java.util.ArrayList;
import java.util.List;

public class task {

    private String taskName;
    private String taskDescription;
    private String sender;
    private String recipient;
    private boolean isCompleted;

    public task(String taskName, String taskDescription, String sender, String recipient) {
        this.taskName = taskName;
        this.taskDescription = taskDescription;
        this.sender = sender;
        this.recipient = recipient;
        this.isCompleted = false;
    }

    public String getTaskName() {
        return taskName;
    }

    public void setTaskName(String taskName) {
        this.taskName = taskName;
    }

    public String getTaskDescription() {
        return taskDescription;
    }

    public void setTaskDescription(String taskDescription) {
        this.taskDescription = taskDescription;
    }

    public String getSender() {
        return sender;
    }

    public void setSender(String sender) {
        this.sender = sender;
    }

    public String getRecipient() {
        return recipient;
    }

    public void setRecipient(String recipient) {
        this.recipient = recipient;
    }

    public boolean isIsCompleted() {
        return isCompleted;
    }

    public void setIsCompleted(boolean isCompleted) {
        this.isCompleted = isCompleted;
    }

    public String getTaskDetails() {
        return "Task Name: " + taskName + ", Description: " + taskDescription + ", Sender: " + sender + ", Recipient: " + recipient + ", Completed: " + isCompleted;
    }

    public static void main(String[] args) {
        // Example Usage
        task myTask = new task("Grocery Shopping", "Buy milk, eggs, and bread", "Alice", "Bob");
        System.out.println(myTask.getTaskDetails());

        myTask.setIsCompleted(true);
        System.out.println(myTask.getTaskDetails());
    }
}

File: code-awesome/java-quarkus-vue/super_dating/frontend/src/pages/task.js
// frontend/src/pages/task.js

// Let's start with a delightful and sophisticated task management app!
// We'll call this 'Super Dating' – because, why not? 🚀

// Our variables will be named after cute animals to make the code more fun!
let adorableTaskName = "";
let cuteTaskDescription = "";
let fuzzyDueDate = null;
let lovableAssignee = null;

// This is the main function that handles the creation of a new task
function createTask() {
  adorableTaskName = document.getElementById("taskName").value;
  cuteTaskDescription = document.getElementById("taskDescription").value;
  fuzzyDueDate = document.getElementById("dueDate").value;

  // Basic input validation (let's be responsible!)
  if (!adorableTaskName || !cuteTaskDescription) {
    alert("Task name and description are required!");
    return;
  }

  // Let's create a task object (a little bit fancy)
  let newTask = {
    name: adorableTaskName,
    description: cuteTaskDescription,
    dueDate: fuzzyDueDate,
    assignee: lovableAssignee,
  };

  // Let's display the new task on the page (a beautiful presentation!)
  displayTask(newTask);

  // Reset the input fields (a clean slate for the next brilliant idea!)
  document.getElementById("taskName").value = "";
  document.getElementById("taskDescription").value = "";
  document.getElementById("dueDate").value = "";
}

// Function to display the task on the page. This is our beautiful presentation!
function displayTask(task) {
  // Create an element to hold the task information
  let taskElement = document.createElement("div");
  taskElement.className = "task";

  // Add the task name
  let taskNameElement = document.createElement("h2");
  taskNameElement.textContent = "Task: " + task.name;
  taskElement.appendChild(taskNameElement);

  // Add the task description
  let taskDescriptionElement = document.createElement("p");
  taskDescriptionElement.textContent = "Description: " + task.description;
  taskElement.appendChild(taskDescriptionElement);

  // Add the due date
  let dueDateElement = document.createElement("p");
  dueDateElement.textContent = "Due Date: " + task.dueDate;
  taskElement.appendChild(dueDateElement);

  // Add the assignee
  let assigneeElement = document.createElement("p");
  assigneeElement.textContent = "Assignee: " + task.assignee;
  taskElement.appendChild(assigneeElement);

  // Append the task element to the main container
  document.getElementById("tasksContainer").appendChild(taskElement);
}

// Event listener for the "Create Task" button
document.getElementById("createTaskButton").addEventListener("click", createTask);

// Initial display of tasks (for demonstration purposes)
displayTask({ name: "Grocery Shopping", description: "Buy milk, eggs, and bread", dueDate: "2024-03-15", assignee: "Alice" });
displayTask({ name: "Schedule Meeting", description: "Discuss project progress", dueDate: "2024-03-16", assignee: "Bob" });